common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -712,6 +794,7 @@ public MergeStatuses finalizeShuffleMerge(FinalizeShuffleMerge msg) {
       mergeStatuses = new MergeStatuses(msg.shuffleId, msg.shuffleMergeId,
         bitmaps.toArray(new RoaringBitmap[bitmaps.size()]), Ints.toArray(reduceIds),
         Longs.toArray(sizes));
+      appShuffleInfo.shuffles.get(msg.shuffleId).setFinalizedPartitions(Ints.toArray(reduceIds));
```

Comment:
The `FinalizedPartitions` will be empty after the Shuffle service restart which will cause the merged shuffle files leak.

Reply:
Yes, if we need to solve this situation completely, we need to store finalized partitions in db.
But on the other hand, when Application removed, all the merged data will be cleaned up finally.

Reply:
What if the shuffle statuses are not exists ?
I think it will not, please see [case match codes](https://github.com/apache/spark/pull/38560/files#diff-21bb40987f7d21c8d3d3de3212f388c210fb04a636c96bdef53cab6070c6aa46R336)